src/services/catalog.service.ts

```diff
@@ -0,0 +1,198 @@
+    return new ServiceItem(resource, this);
+  }
+
+  public getImageItem(resource: any) {
+    return new ImageItem(resource, this);
+  }
+
+  public getCategoryFromSubCategory(subCategory: string) {
+    let retCategory: string = 'other';
+
+    let breakLoops: boolean = false;
+    _.each(this.categories, (category) => {
+      if (breakLoops) {
+        return false;
+      }
+      _.each(category.subCategories, (cat_subCategory) => {
+        if (cat_subCategory.id === subCategory) {
+          retCategory = category.id;
+          breakLoops = true;
+          return false;
+        }
+      });
+    });
+
+    return retCategory;
+  }
+
+  public getSubCategoryFromAlias(subCategory: string) {
+    let retSubCategory: string = subCategory;
+
+    let breakLoops: boolean = false;
+    _.each(this.categories, (category) => {
+      if (breakLoops) {
+        return false;
+      }
+      _.each(category.subCategories, (cat_subCategory) => {
+        if (cat_subCategory.categoryAliases) {
+          let result = _.findIndex(cat_subCategory.categoryAliases, (alias) => {
+            return alias === subCategory;
+          });
+          if (result !== -1) {
+            retSubCategory = cat_subCategory.id;
+            breakLoops = true;
+            return false;
+          }
+        }
+      });
+    });
+    return retSubCategory;
+  }
+}
+
+interface IServiceItem {
+  iconClass: string;
+  imageUrl: string;
+  name: string;
+  catsBySubCats: any;
+  description: string;
+  longDescription: string;
+  resource: any;
+}
+
+export class ServiceItem implements IServiceItem {
+  public iconClass: string;
+  public imageUrl: string;
+  public name: string;
+  public catsBySubCats: any;
+  public description: string;
+  public longDescription: string;
+  public resource: any;
+  private catalogSrv: CatalogService;
+
+  constructor (serviceClass: any, catalogSrv: CatalogService) {
+    this.resource = serviceClass;
+    this.catalogSrv = catalogSrv;
+    this.imageUrl = this.getImage();
+    this.name = this.getName();
+    this.description = this.getDescription();
+    this.longDescription = this.getLongDescription();
+    this.catsBySubCats = this.getCategoriesBySubCategories();
+  }
+
+  private getImage() {
+    if (this.resource.metadata) {
+      return this.resource.metadata.imageUrl;
+    }
+  }
+
+  private getName() {
+    if (this.resource.osbMetadata) {
+      return this.resource.osbMetadata.displayName ? this.resource.osbMetadata.displayName : this.resource.metadata.name;
+    } else {
+      return this.resource.metadata.name;
+    }
+  }
+
+  private getDescription() {
+    if (this.resource.osbMetadata) {
+      return this.resource.osbMetadata.description;
+    }
+  }
+
+  private getLongDescription() {
+    if (this.resource.osbMetadata) {
+      return this.resource.osbMetadata.longDescription;
+    }
+  }
+
+  private getCategoriesBySubCategories() {
+    let catsBySubCats = {};
+
+    if (this.resource.osbTags) {
```

Comment:
the if isn't necessary, _.each already handles existence checks on what you pass to it so if this.resource.osbTags is null or undefined it won't care it just treats it like there are no items